module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h

```diff
@@ -54,15 +54,15 @@ namespace behaviour {
             //-----------non-config variables (not defined in WalkPathPlanner.yaml)-----------
 
             // info for the current walk
-            arma::vec2 currentTargetPosition;
-            arma::vec2 currentTargetHeading;
+            Eigen::Vector2d currentTargetPosition;
+            Eigen::Vector2d currentTargetHeading;
             message::behaviour::KickPlan targetHeading;
-            arma::vec2 targetPosition = {0, 0};
+            Eigen::Vector2d targetPosition = Eigen::Vector2d::Zero();
 
             NUClear::clock::time_point timeBallLastSeen;
-            arma::vec3 rBWw          = {10, 0, 0};
-            bool robot_ground_space  = true;
-            arma::vec2 position      = {1, 0};  // ball pos rel to robot
+            Eigen::Vector3d rBWw{Eigen::Vector3d(10.0, 0.0, 0.0)};
+            bool robot_ground_space = true;
+            Eigen::Vector2d position{Eigen::Vector2d(Eigen::Vector2d::UnitX())};  // ball pos rel to robot
```

Comment:
The one here is the worst of both worlds, should be

`Eigen::Vector3d rBWw(10.0, 0.0, 0.0);`

Reply:
I agree, but when I tried to convert these to that style, I got these errors (I think because this is in a header file?)

```sh
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h:60:44: error: ‘Eigen::Vector2d<double, 2, 1, 0, 2, 1>::Zero’ is not a type
   60 |             Eigen::Vector2d targetPosition(Eigen::Vector2d::Zero());

.
.
.

In file included from /home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp:20:
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h:63:50: error: expected ‘)’ before numeric constant
   63 |             Eigen::Vector3d rBWw(Eigen::Vector3d(10.0, 0.0, 0.0));
      |                                                 ~^~~~
      |                                                  )
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h:63:50: error: expected ‘)’ before numeric constant
   63 |             Eigen::Vector3d rBWw(Eigen::Vector3d(10.0, 0.0, 0.0));
      |                                 ~                ^~~~
      |                                                  )
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h:63:49: error: expected ‘;’ at end of member declaration
   63 |             Eigen::Vector3d rBWw(Eigen::Vector3d(10.0, 0.0, 0.0));
      |                                                 ^
# etc

.
.
.

In file included from /home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp:20:
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.h:65:38: error: ‘Eigen::Vector2d<double, 2, 1, 0, 2, 1>::UnitX’ is not a type
   65 |             Eigen::Vector2d position(Eigen::Vector2d::UnitX());  // ball pos rel to robot
      |                                      ^~~~~
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp: In lambda function:
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp:199:21: error: invalid use of member function ‘Eigen::Vector2d module::behaviour::planning::SimpleWalkPathPlanner::position(int (*)())’ (did you forget the ‘()’ ?)
  199 |                     position            = pos.head<2>();
      |                     ^~~~~~~~
/home/nubots/NUbots/module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp:214:78: error: invalid use of non-static member function ‘Eigen::Vector2d module::behaviour::planning::SimpleWalkPathPlanner::position(int (*)())’
  214 |                         Eigen::Vector2d ballToTarget = (kickTarget.head<2>() - position).normalized();
# etc
```
So I think it's either of these ones, or some other workaround

Reply:
Being a header file will never have any influence on an error. C++ can't tell the difference between a header file and a c++ file. I don't have enough of the error to see what's happening there though. Also, that's still a constructor calling a constructor.

`Eigen::Vector3d rBWw(Eigen::Vector3d(10.0, 0.0, 0.0));` should be `Eigen::Vector3d rBWw(10.0, 0.0, 0.0);`

Reply:
hmm, I meant like, in a class and not within a method, rather than in a `.h` I guess. I don't think it is one of those errors though. I tried each one separately  and retried each one separately when I added the comment with the errors. If it is an error like that I am going to have to retire from coding 😂

Reply:
Ahh wait, yes I see it now, GitHub wasn't giving me enough context. Yeah, you don't do constructors like that when they're just values in the H file. Instead, it really should be to be an = sign.

```cpp
Eigen::Vector3d rBWw = Eigen::Vector3d(10.0, 0.0, 0.0);
Eigen::Vector2d position = Eigen::Vector2d::UnitX();
```